src/tls12-download.c

```diff
@@ -326,10 +329,58 @@ int __stdcall entry()
     WinHttpCloseHandle(request);
     WinHttpCloseHandle(connect);
     WinHttpCloseHandle(session);
-    CloseHandle(out_file);
 
-    write_message(std_out, L"Done.\r\n");
+    write_message(std_out, L" done.\r\nValidating signature...");
+
+    if (SetFilePointer(out_file, 0, 0, FILE_BEGIN) == INVALID_SET_FILE_POINTER)
+    {
+        abort_api_failure(std_out, L"SetFilePointer");
+    }
+
+    DWORD exit_code = 0;
+
+    WINTRUST_FILE_INFO wtfi;
+    wtfi.cbStruct = sizeof(wtfi);
+    wtfi.pcwszFilePath = out_file_path;
+    wtfi.hFile = out_file;
+    wtfi.pgKnownSubject = 0;
+
+    WINTRUST_DATA wtd;
+    ZeroMemory(&wtd, sizeof(wtd));
+    wtd.cbStruct = sizeof(wtd);
+    wtd.dwUIChoice = WTD_UI_NONE;
+    wtd.fdwRevocationChecks = WTD_REVOKE_WHOLECHAIN;
+    wtd.dwUnionChoice = WTD_CHOICE_FILE;
+    wtd.pFile = &wtfi;
+    wtd.dwStateAction = WTD_STATEACTION_VERIFY;
+    wtd.dwProvFlags = WTD_REVOCATION_CHECK_CHAIN;
+
+    GUID wt_policy_guid = WINTRUST_ACTION_GENERIC_VERIFY_V2;
+
+    LONG trust_validation_result = WinVerifyTrust(INVALID_HANDLE_VALUE, &wt_policy_guid, &wtd);
+    wtd.dwStateAction = WTD_STATEACTION_CLOSE;
+    (void)WinVerifyTrust(INVALID_HANDLE_VALUE, &wt_policy_guid, &wtd);
+    if (trust_validation_result == 0)
```

Comment:
```suggestion
    if (trust_validation_result == ERROR_SUCCESS)
```

Reply:
It's documented to return "zero", not ERROR_SUCCESS: https://docs.microsoft.com/en-us/windows/win32/api/wintrust/nf-wintrust-winverifytrust#return-value

Reply:
Thanks, comprehension fail on my part. I didn't see that line and looked to the example[1] to see what success looks like (which uses `ERROR_SUCCESS`).

[1] https://docs.microsoft.com/en-us/windows/win32/seccrypto/example-c-program--verifying-the-signature-of-a-pe-file